modules/workergroup/instanceconfig.tf

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+resource "oci_core_instance_configuration" "instance_configuration" {
+  # Create an OCI Instance Configuration resource for each enabled entry of the worker_groups map with a mode that uses one.
+  for_each       = local.enabled_instance_configs
+  compartment_id = each.value.compartment_id
+  display_name   = "${each.value.label_prefix}-${each.key}"
+
+  instance_details {
+    instance_type = "compute"
+
+    launch_details {
+      # Define each configured availability domain for placement, with bounds on # available
+      # Configured AD numbers e.g. [1,2,3] are converted into tenancy/compartment-specific names
+      availability_domain = lookup(local.ad_number_to_name, (
```

Comment:
The placement for instance config is different from that of node pool. Unless there's a reason behind this, we should be consistent

Reply:
The definition on the pool is what matters, and this is optional - removing.